### PR TITLE
WRO-2580: Add optional `containerOption.toOuterContainer` to `focus` function

### PR DESCRIPTION
**Pull request by jeonghee27**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
I need a api that iteratively search the outer container to give it focus.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Added `containerOption.toOuterContainer` option to `focus` function

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-19179

### Comments
Enact-DCO-1.0-Signed-off-by: Jeonghee Ahn (jeonghee27.ahn@lge.com)

### CODE REVIEW

File: packages/spotlight/src/spotlight.js [hunk 1]
```diff
@@ -774,6 +774,32 @@ const Spotlight = (function () {
...
+		// focusFromContainer()
+		// focusFromContainer(<containerId>)
+		/**
+		 * Focuses from the specified container.
+		 * If the proper target is not in the container, repeat with the outer container.
+		 *
+		 * If Spotlight is in pointer mode, focus is not changed but `elem` will be set as the last
+		 * focused element of its spotlight containers.
+		 *
+		 * @param {String} [containerId] The starting spotlight container ID . If not supplied,
```
Comment (MikyungKim) [inline packages/spotlight/src/spotlight.js:786]:
```suggestion
		 * @param {String} [containerId] The starting spotlight container ID. If not supplied,
```

File: packages/spotlight/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 The following is a curated list of changes in the Enact spotlight module, newest changes on the top.
 
+## [unreleased]
+
+### Added
+
+- `spotlight.Spotlight` method `focusFromContainer()` to focus recursively
```
Comment (MikyungKim) [inline packages/spotlight/CHANGELOG.md:9]:
```suggestion
- `spotlight` function `focusFromContainer` to focus from a container to outer container recursively
```

**Comment by MikyungKim** [comment]
Please update the branch with the latest develop.

File: packages/spotlight/src/spotlight.js (hunk 1 above)
Comment (jeonghee27) [reply packages/spotlight/src/spotlight.js]: I fixed

File: packages/spotlight/CHANGELOG.md (hunk 2 above)
Comment (jeonghee27) [reply packages/spotlight/CHANGELOG.md]: I fixed

**Review by MikyungKim — COMMENTED** [review]
Let's discuss how we expose the API either adding an option to existing method or adding the new one.

File: packages/spotlight/src/spotlight.js [hunk 3]
```diff
@@ -730,21 +730,27 @@ const Spotlight = (function () {
...
 		 * @public
 		 */
 		focus: function (elem, containerOption = {}) {
 			let target = elem;
 			let wasContainerId = false;
+			let currentContainerNode;
 
 			if (!elem) {
 				target = getTargetByContainer();
+				currentContainerNode = getContainerNode(getLastContainer());
```
Comment (MikyungKim) [inline packages/spotlight/src/spotlight.js:748]: If we don't support selector type, let's remove this too. And let's add some docs for the constraints.

File: packages/spotlight/src/spotlight.js [hunk 4]
```diff
@@ -730,21 +730,27 @@ const Spotlight = (function () {
...
 		 *  component or a spotlight container, or spottable node. If not supplied, the default
 		 *  container will be focused.
-		 * @param {Object} [containerOption] An optional object containing preferred `enterTo`.
-		 *  It will be passed to the `getTargetByContainer` when `elem` is a container.
+		 * @param {Object} [containerOption] An optional The object including `enterTo` and `toOuterContainer`.
+		 * @param {String} [containerOption.enterTo] It will be passed to the `getTargetByContainer`
+		 *  when `elem` is a container.
+		 * @param {Boolean} [containerOption.toOuterContainer] If the proper target is not found, search one
+		 *  recursively to outer container.
+		 * It works when the first parameter ` elem ` is either a spotlight container ID or a spotlight container node.
```
Comment (MikyungKim) [inline packages/spotlight/src/spotlight.js:738]:
```suggestion
		 * It works when the first parameter `elem` is either a spotlight container ID or a spotlight container node.
```

**Review by MikyungKim — APPROVED** [review]
LGTM

File: packages/spotlight/src/spotlight.js [hunk 5]
```diff
@@ -730,21 +730,27 @@ const Spotlight = (function () {
...
+		 *  `enterTo` configuration. It will be passed to the `getTargetByContainer`.
+		 * @param {Boolean} [containerOption.toOuterContainer] If the proper target is not found, search one
+		 *  recursively to outer container.
 		 * @returns {Boolean} `true` if focus successful, `false` if not.
 		 * @public
 		 */
 		focus: function (elem, containerOption = {}) {
 			let target = elem;
 			let wasContainerId = false;
+			let currentContainerNode;
```
Comment (0x64) [inline packages/spotlight/src/spotlight.js:745]: What about to set an initial value of `null` as this variable's value could never be assigned in some condition?

File: packages/spotlight/src/spotlight.js [hunk 6]
```diff
@@ -730,21 +730,27 @@ const Spotlight = (function () {
 		 * @param {String|Node} [elem] The spotlight ID or selector for either a spottable
 		 *  component or a spotlight container, or spottable node. If not supplied, the default
 		 *  container will be focused.
-		 * @param {Object} [containerOption] An optional object containing preferred `enterTo`.
-		 *  It will be passed to the `getTargetByContainer` when `elem` is a container.
+		 * @param {Object} [containerOption] The object including `enterTo` and `toOuterContainer`.
+		 *  It works when the first parameter `elem` is either a spotlight container ID or a spotlight container node.
+		 * @param {('last-focused'|'default-element'|'topmost')} [containerOption.enterTo] Specifies preferred
+		 *  `enterTo` configuration. It will be passed to the `getTargetByContainer`.
```
Comment (0x64) [inline packages/spotlight/src/spotlight.js]: Let's remove the later sentence to avoid any confusion of app devs.

**Pull request merged**